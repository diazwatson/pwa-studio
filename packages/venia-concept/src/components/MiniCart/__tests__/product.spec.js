import React, { useState } from 'react';
import ShallowRenderer from 'react-test-renderer/Shallow';

import Product from '../product';

global.getComputedStyle = jest.fn().mockReturnValue({
    getPropertyValue: jest.fn().mockReturnValue('80px')
});
jest.mock('react', () => {
    const React = jest.requireActual('react');
    const spy = jest.spyOn(React, 'useState');

    return Object.assign(React, { useState: spy });
});

const renderer = new ShallowRenderer();

const props = {
    beginEditItem: jest.fn(),
    currencyCode: 'US',
    item: {
        image: {
            file: 'unittest'
        },
        name: 'Unit Test Product',
        options: [],
        price: 99,
        qty: 1
    },
    removeItemFromCart: jest.fn()
};

test('it renders correctly', () => {
    const tree = renderer.render(<Product {...props} />);

    expect(tree).toMatchSnapshot();
});

test('it renders a mask div before the kebab if it is loading', () => {
    useState
        // backgroundImage
        .mockReturnValueOnce([null, jest.fn()])
        // isFavorite
        .mockReturnValueOnce([false, jest.fn()])
        // isLoading
        .mockReturnValueOnce([true, jest.fn()]);

    const tree = renderer.render(<Product {...props} />);

    expect(tree).toMatchSnapshot();
});
