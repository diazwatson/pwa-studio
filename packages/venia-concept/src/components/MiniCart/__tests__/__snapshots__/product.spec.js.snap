// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a mask div before the kebab if it is loading 1`] = `
<li>
  <div
    style={
      Object {
        "backgroundImage": null,
      }
    }
  />
  <div>
    Unit Test Product
  </div>
  <ProductOptions
    options={Array []}
  />
  <div>
    <div>
      <span>
        1
      </span>
      <span>
        ×
      </span>
      <span>
        <mockConstructor
          currencyCode="US"
          value={99}
        />
      </span>
    </div>
  </div>
  <div />
  <Kebab>
    <Section
      icon="Heart"
      iconAttributes={Object {}}
      onClick={[Function]}
      text="Add to favorites"
    />
    <Section
      icon="Edit2"
      onClick={[Function]}
      text="Edit item"
    />
    <Section
      icon="Trash"
      onClick={[Function]}
      text="Remove item"
    />
  </Kebab>
</li>
`;

exports[`it renders correctly 1`] = `
<li>
  <div
    style={
      Object {
        "backgroundImage": null,
      }
    }
  />
  <div>
    Unit Test Product
  </div>
  <ProductOptions
    options={Array []}
  />
  <div>
    <div>
      <span>
        1
      </span>
      <span>
        ×
      </span>
      <span>
        <mockConstructor
          currencyCode="US"
          value={99}
        />
      </span>
    </div>
  </div>
  <Kebab>
    <Section
      icon="Heart"
      iconAttributes={Object {}}
      onClick={[Function]}
      text="Add to favorites"
    />
    <Section
      icon="Edit2"
      onClick={[Function]}
      text="Edit item"
    />
    <Section
      icon="Trash"
      onClick={[Function]}
      text="Remove item"
    />
  </Kebab>
</li>
`;
